File: 0234-palindrome-linked-list/0234-palindrome-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function isPalindrome(head: ListNode | null): boolean {
    if(!head && !head.next) return true
    let str = ""
    while(head){
        str += head.val;
        head = head.next;
    }
    let x = str.split('').reverse().join("");
    return x == str
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next: return True
        astr = ''
        while head:
            astr += str(head.val)
            head = head.next
        print(astr)
        x = astr[::-1]
        print(x)
        
        return x == astr
        